.github/workflows/release.yml

```
@@ -1,82 +1,100 @@
-name: Release
+name: Main Release
 
 on:
-  workflow_dispatch:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+
+permissions:
+  contents: write
 
 jobs:
-  build:
+  release:
+    if: github.event.pull_request.merged == true
     runs-on: macos-latest
 
     steps:
-      - name: Checkout Repository
+      - name: Checkout code
         uses: actions/checkout@v5
         with:
           token: ${{ secrets.LICHTBLICK_GITHUB_TOKEN }}
 
       - name: Set up Node.js
-        uses: actions/setup-node@v4.0.3
+        uses: actions/setup-node@v4
         with:
-          node-version: 16.17
+          node-version: 18
 
       - name: Enable corepack
         run: corepack enable yarn
 
-      - name: Bump minor version and save it on environment variable
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Determine version bump type
+        id: bump_type
         run: |
-          yarn version minor
-          NEW_VERSION=$(node -p "require('./package.json').version")
-          echo "new_version=$NEW_VERSION" >> $GITHUB_ENV
+          branch_name="${GITHUB_HEAD_REF}"
+          if [[ "$branch_name" == hotfix/* ]]; then
+            echo "type=patch" >> $GITHUB_ENV
+          elif [[ "$branch_name" == release/* ]]; then
+            echo "type=minor" >> $GITHUB_ENV
+          else
+            echo "type=patch" >> $GITHUB_ENV
+          fi
```

In theory, release/* branches can be both patch, minor or major. One way to identify what type of release it will be is via branch name (release/major, release/minor, release/patch).
I don't remember well if we discussed about it, but we can also discuss it further and leave it for later. Since this approach is simple and suits us for now.

```suggestion
        run: |
          branch_name="${GITHUB_HEAD_REF}"
          if [[ "$branch_name" == hotfix/* ]]; then
            echo "type=patch" >> $GITHUB_ENV
          elif [[ "$branch_name" =~ ^release/major ]]; then
            echo "type=major" >> $GITHUB_ENV
          elif [[ "$branch_name" =~ ^release/minor ]]; then
            echo "type=minor" >> $GITHUB_ENV
          elif [[ "$branch_name" =~ ^release/patch ]]; then
            echo "type=patch" >> $GITHUB_ENV
          elif [[ "$branch_name" == release/* ]]; then
            # Default para release branches sem especificação
            echo "type=minor" >> $GITHUB_ENV
          else
            echo "type=patch" >> $GITHUB_ENV
          fi
```